addons/account/models/account_journal.py

```diff
@@ -484,6 +484,31 @@ def _check_auto_post_draft_entries(self):
     def _onchange_type(self):
         self.refund_sequence = self.type in ('sale', 'purchase')
 
+    @api.onchange('payment_sequence')
+    def _onchange_payment_sequence(self):
+        if self.payment_sequence:
+            existing_payment_moves = self.env['account.move'].search([('journal_id', '=', self._origin.id)]).filtered(lambda m: m.payment_id)
+            existing_stmt_line_moves = self.env['account.move'].search([('journal_id', '=', self._origin.id)]).filtered(lambda m: m.statement_line_id)
+            if not existing_stmt_line_moves and existing_payment_moves and any(move.sequence_prefix.startswith(self.code) for move in existing_payment_moves):
+                return {
+                    'warning': {
+                        'title': _("Warning!"),
+                        'message': _("You have certain payments with a sequence prefix that starts with %s. This can cause conflicts with new statement lines.\n"
+                                     "It is advisable to consider changing the sequence prefix of these payments by resequencing them.", self.code),
+                    }
+                }
+            elif existing_payment_moves and existing_stmt_line_moves:
+                payment_move_prefix = existing_payment_moves[-1].sequence_prefix.split('/')[0]
+                stmt_move_prefix = existing_stmt_line_moves[-1].sequence_prefix.split('/')[0]
```

Comment:
Same problem as above I think, no ? ^^

Reply:
I'll check that

Reply:
In fact, if there are both statement lines and payment moves, only the last of each matters

Reply:
ok, let's assume 3 payment and 3 stmt line, you can change the sequence of any of those, the fact is if the last statement line's move and the last payment's move have the same sequence, we will continue to have the same sequence, regardless the fact we set the dedicated payment sequence on the journal, and if the last move of each(payment and statement line) have different sequence because of resequencing, we have already dedicated payement sequence even before setting the option, then no need to display a warning.

Reply:
3 statement lines: 

- MISC/2023/00001
- MISC/2023/00002
- MISC/2023/00003 -> change it for MISC2/2023/00001

3 payments:

- MISC/2023/00004
- MISC/2023/00005
- MISC/2023/00006 -> change it to MISC2/2023/00002

Activate Dedicated Payment Sequence on the journal and make a new payment: it will have the sequence MISC2/2023/00003, make a new statement line, move will have MISC2/2023/00004

Now do that:

3 statement lines: 

- MISC/2023/00001
- MISC/2023/00002
- MISC/2023/00003

3 payments:

- MISC/2023/00004
- MISC/2023/00005
- MISC/2023/00006 -> change it to MISC2/2023/00001

Activate Dedicated Payment Sequence on the journal and make a new payment: it will have the sequence MISC2/2023/00002, make a new statement line, move will have MISC/2023/00006